clang/test/ParserHLSL/hlsl_annotations_on_struct_members.hlsl

```diff
@@ -0,0 +1,30 @@
+// RUN: %clang_cc1 -triple dxil-pc-shadermodel6.0-compute -x hlsl -o - %s -verify

```

Comment:
Could we add another test which compiles successfully and check the AST?

Reply:
There are currently no candidate attributes that have subjects that are field decls other than SV_DispatchThreadID. In other words, SV_DispatchThreadID is the only attribute that appertains to FieldDecls, all other attributes will fail the `diagAppertainsTo` check inside of `tools\clang\include\clang\Sema\AttrParsedAttrImpl.inc`, and that test needs to pass before the parsed attribute can be applied to the decl, and thus appear on the AST.